gen/build_deploy/azure.py

```diff
@@ -43,13 +43,21 @@ def validate_provider(provider):
     assert provider == 'azure'
 
 
+def calculate_public_route_to_master_available(oauth_available, custom_auth):
+    if oauth_available == 'true' or custom_auth == 'true':
+        return 'true'
+    else:
+        return 'false'
```

Comment:
Its more that if you decide to roll your own, you can just set your config preferences to be oauth_availble = "false" in which case the generated template will have no auth in DC/OS and an HTTP/S route to master. There is no guarantee that "custom_auth" is true if oauth is unavailable